### ISSUE TITLE
configure of synthesis_tt.py is  mismatch  train_tt.py

### ISSUE THREAD

**Issue by shanguanma** [uwjunqi/Tensor-Train-Neural-Network#1]
Hi, I am reading your newest paper(e.g: "TENSOR-TO-VECTOR REGRESSION FOR MULTI-CHANNEL SPEECH ENHANCEMENT BASED ON TENSOR-TRAIN NETWORK", "Exploring Deep Hybrid Tensor-to-Vector Network Architectures for Regression-Based Speech Enhancement") and running your publish code, I have a question for you. it  is as follows: 
1. I found that the network structure of synthesis_tt.py was different from train_tt.py, so it is failed for the synthesis stage.
can you fix this? thanks.

**Comment by jqi41** [comment]
Hello, the code `synthesis_tt.py' has been updated.

**Comment by huckiyang** [comment]
@shanguanma thank you for your interests and feel free to let us know if the problem is yet fixed.

**Comment by shanguanma** [comment]
Thanks for your quick reply. @uwjunqi @huckiyang. 
I will test it later.
However, I have a question. in the "train_tt.py".
`parser.add_argument('--hidden_tensors', default=[[8, 4, 8, 4], [8, 4, 8, 4], [8, 4, 8, 8]])`
`tt_rank = [1, 2, 2, 2, 1]`
In the "synthesis_tt.py",
`parser.add_argument('--tt_rank', default=[1, 3, 1], help='tensor-train rank')`
`parser.add_argument('--hidden_tensors', default=[[80, 100], [80, 100], [80, 100], [80, 100], [160, 100]], help='hidden tensors')`
Why is different between train stage and inference stage ?

**Comment by shanguanma** [comment]
I have test " synthesis_tt.py",
I used the command line is as follows:
`$ python3 synthesis_tt.py  --noise_list_fn data/noisy_testset_wav_16k/wav_list  --load_model_path exp/dnn_tt_56spk/v3_raw/model_4.pkl `

But get the below error,
`/home4/md510/w2020/espnet_20200719/espnet/tools/venv/lib/python3.7/site-packages/librosa/util/decorators.py:9: NumbaDeprecationWarning: An import was requested from a module that has moved location.
Import requested from: 'numba.decorators', please update to use 'numba.core.decorators' or pin to Numba version 0.48.0. This alias will not be present in Numba version 0.50.0.
  from numba.decorators import jit as optional_jit
/home4/md510/w2020/espnet_20200719/espnet/tools/venv/lib/python3.7/site-packages/librosa/util/decorators.py:9: NumbaDeprecationWarning: An import was requested from a module that has moved location.
Import of 'jit' requested from: 'numba.decorators', please update to use 'numba.core.decorators' or pin to Numba version 0.48.0. This alias will not be present in Numba version 0.50.0.
  from numba.decorators import jit as optional_jit
usage: synthesis_tt.py [-h] [--batch_size BATCH_SIZE]
                       [--input_tensor INPUT_TENSOR]
                       [--hidden_tensors HIDDEN_TENSORS] [--data_path DIR]
                       [--n_epochs N_EPOCHS]
synthesis_tt.py: error: unrecognized arguments: --noise_list_fn data/noisy_testset_wav_16k/wav_list --load_model_path exp/dnn_tt_56spk/v3_raw/model_4.pkl
`

**Comment by shanguanma** [comment]
Dear @uwjunqi @huckiyang , can you help me to fix this problem?

**Comment by jqi41** [comment]
@shanguanma You have to keep a consistent TT-DNN during the training and synthesizing process. I have adjusted the defaulted parameters such that you may find it easier to implement. As for another problem, I have no idea about the Numba issue you encountered because it is related to the Python version you used.

**Comment by shanguanma** [comment]
@uwjunqi, Thanks for your reply. Yes, I will update it and test it.

**Comment by shanguanma** [comment]
@uwjunqi, The above error shows that running synthesis_tt.py throws the parameter list of train_tt.py. This problem is because the input latitude of your network depends on the characteristics of the data, and this data characteristic depends on the latitude of your training data, but in the inference stage, When you import the model again, the parameter latitude is not entered, so the above error appears. this problem has solved it by myself.
I found that output dim=257  in updated "synthesis_tt.py", however  output dimension the last layer of "train_tt.py" is 256. I think they are a mismatch. it should be a bug.

**Comment by shanguanma** [comment]
@uwjunqi, The above error shows that running synthesis_tt.py throws the parameter list of train_tt.py. This problem is because the input dimension of your network depends on the feature of the train data, but in the inference stage, When you import the model again, the parameter of the input dimension is not entered, so the above error appears. This problem has solved it by myself.
I found that the output dimension is 257 in updated "synthesis_tt.py", however output dimension the last layer of "train_tt.py" is 256. I think they are a mismatch. it should be a bug.

**Comment by huckiyang** [comment]
It is a librosa version issue. Please check the error msg directly. https://github.com/librosa/librosa/issues/1155
